base/loading.jl

```diff
@@ -2,6 +2,84 @@
 
 # Base.require is the implementation for the `import` statement
 
+# Cross-platform case-sensitive path canonicalization
+
+if OS_NAME ∈ (:Linux, :FreeBSD)
+    # Case-sensitive filesystems, don't have to do anything
+    isfile_casesensitive(path) = isfile(path)
+elseif OS_NAME == :Windows
+    # GetLongPathName Win32 function returns the case-preserved filename on NTFS.
+    function isfile_casesensitive(path)
+        isfile(path) || return false  # Fail fast
+        longpath(path) == path
```

Comment:
Possibly this should be `basename(longpath(path)) == basename(path)`?  Two reasons:

* On MacOS, we only check that the basename matches, so this is inconsistent.
* The intent of this was that `import Foo` would check for `Foo` in a case-sensitive way.  But this does more than that, and checks the whole path.  This means that users `LOAD_PATH` and `HOME` variables becomes case-sensitive, for example, which is unexpected on case-insensitive systems, and has led to at least one [confused user on discourse](https://discourse.julialang.org/t/windows-capitalization-of-the-home-environment-variable/200?u=stevengj).